docs/Troubleshooting.md

```diff
@@ -311,3 +311,19 @@ import * as React from 'react' // React is now patched
 import React from 'react'
 import { hot } from 'react-hot-loader' // React is now patched
 ```
+
+#### React-hot-loader: an instance were returned from a render function. Configuration change required
+
+React-hot-loader runs in "compatibility" mode by default. This disables some workarounds you possible might need.
+
+* **Enable** compat mode if you are using something like `react-async-bootstrapper`(react-async-component)
+* **Disable** compat mode if you are using `Relay`.
+
+There is **no** way to use them simultaneously.
+
+Buy default RHL will produce more simply and compatible code, but non-compact mode is absolutely legit for React 15+.
```

Comment:
This is not very clear, I suggest to change everything and put it in the README :

```
#### Using React Hot Loader + Relay

To enable Relay compatibility mode, you must define a custom config:

__```js
import { setConfig } from 'react-hot-loader'
setConfig({ relayCompatMode: true })
__```

This mode is not compatible with `react-async-bootstrapper` or `react-async-component`, more detail in #833.
```

Reply:
We do know that "the hack" is needed only for Relay, but it is an absolutely legit structure from React prospective, and someone else could use it. Thus we should not call it `relay` even if we don't know that someone.

Reply:
I agree, but could we find a name more precise than `compat`?

Reply:
Maybe long and explicit, it is OK for me.

Reply:
the "real" name is `supportIndeterminateComponent`, but actually not support, but "use".
`statelessIndeterminateComponent` ?